packages/core/contracts/proxy-scripts/lsp-broker/LspUniswapV2Broker.sol

```diff
@@ -15,27 +18,197 @@ import "../../financial-templates/long-short-pair/LongShortPair.sol";
+            "Create invariant failed"
+        );
+
+        {
+            // 2) Calculate if we need to sell the long token for the short or the short token for the long. Calculate
+            // the total trade size to bring the balance of long/short in this contract to equal the pool ratio after
+            // the trade. Note this computation must consider the impact of the trade with the resultant pool ratios.
+            bool aToB;
+            uint256 tradeSize;
+            (uint256 reserveA, uint256 reserveB) =
+                getReserves(router.factory(), address(longToken), address(shortToken));
+            (aToB, tradeSize) = computeSwapToMintAtPoolRatio(
+                FixedPoint.Signed(int256(tokensToCreate)),
+                FixedPoint.Signed(int256(reserveA)),
+                FixedPoint.Signed(int256(reserveB))
+            );
+            address[] memory path = new address[](2);
+
+            // 2.a) If the trade is token a to b then we are selling long tokens for short tokens. In this case, we
+            // know the exact output number of tokens we want (short tokens) and use the router's
+            // swapTokensForExactTokens with the exact output number of tokens specified.
+            if (aToB && tradeSize > 0) {
+                path[0] = address(longToken);
+                path[1] = address(shortToken);
+                TransferHelper.safeApprove(address(longToken), address(router), tradeSize);
+                router.swapTokensForExactTokens(tradeSize, type(uint256).max, path, address(this), deadline);
+            }
+            if (!aToB && tradeSize > 0) {
+                // Else, if the trade is b to a then we are selling short tokens for long tokens. In this case, we
+                // know the exact input number of tokens that we need to sell (short tokens) and can use the router's
+                // swapExactTokensForTokens with the exact input spesificed.
+                path[0] = address(shortToken);
+                path[1] = address(longToken);
+                TransferHelper.safeApprove(address(shortToken), address(router), tradeSize);
+                router.swapExactTokensForTokens(tradeSize, 0, path, address(this), deadline);
+            }
+        }
+
+        // 3) Add lequidity to the pool via the router. Approve both long and short tokens with the full account balance
+        // in this contract (i.e the amount of tokens minted +- that traded in step 2.).
+        TransferHelper.safeApprove(address(longToken), address(router), longToken.balanceOf(address(this)));
+        TransferHelper.safeApprove(address(shortToken), address(router), shortToken.balanceOf(address(this)));
+        router.addLiquidity(
+            address(longToken),
+            address(shortToken),
+            longToken.balanceOf(address(this)),
+            shortToken.balanceOf(address(this)),
+            0,
+            0,
+            address(this),
+            deadline
+        );
```

Comment:
I think after this step, we should run a `require(lpToken.balanceOf(address(this)) > minLpTokens` where `uint256 minLpTokens` is a function input that the user specifies to ensure that trade slippage does not result in fewer LP tokens than would normally be mintable with the long-short tokens.

Reply:
ye, not a bad idea. will add it in + associated unit tests.

Reply:
I added this in plus unit tets to validate it. I also added tests to validate `Invalid params` as we have with the swap logic